src/main/java/com/frameworkium/core/ui/pages/Visibility.java

```diff
@@ -58,8 +59,20 @@ public Visibility(Wait<WebDriver> wait, JavascriptExecutor driver) {
      */
     public void waitForAnnotatedElementVisibility(Object pageObject) {
 
-        Field[] allFields = pageObject.getClass().getDeclaredFields();
-        Arrays.stream(allFields)
+        Class <? extends Object> clazz = pageObject.getClass();
+        // Get the declared fields from the current class
+        final List<Field> allFields = new ArrayList<Field>(Arrays.asList(clazz.getDeclaredFields()));
+
+        // Get any declared fields from super classes
+        // i.e. when a page object extends a custom class which itself extends HtmlElement
+        for(clazz = clazz.getSuperclass(); 
+            ((clazz != null) && (clazz != BasePage.class) && (clazz != HtmlElement.class));
+            clazz = clazz.getSuperclass()) {
+            Stream.of(clazz.getDeclaredFields()).
+                    forEach(f->allFields.add(f));
```

Comment:
I think it might be easier to read if we don't modify the `allFields` variable. Perhaps create a new variable called `superClassFields` and then do `Stream.concat(allFields, superClassFields)` below (line 75)?

Reply:
It might be even nicer to make this a private method which takes the class and returns a list of all the fields to keep this method short and readable.

Reply:
Changes made as requested